src/hooks/useTokenOrPairNativeCurrency.ts

```diff
@@ -0,0 +1,72 @@
+import { ChainId, Token, CurrencyAmount, Pair } from '@swapr/sdk'
+
+import { useMemo } from 'react'
+import { useActiveWeb3React } from '.'
+import { useNativeCurrency } from './useNativeCurrency'
+import { usePairLiquidityTokenTotalSupply } from '../data/Reserves'
+import { usePairReserveNativeCurrency } from './usePairReserveNativeCurrency'
+import { useTokenDerivedNativeCurrency } from './useTokenDerivedNativeCurrency'
+import { getLpTokenPrice } from '../utils/prices'
+import { ethers } from 'ethers'
+import Decimal from 'decimal.js-light'
+
+export function useTokenOrPairNativeCurrency(
+  tokenOrPair?: Token | Pair
+): { loading: boolean; derivedNativeCurrency: CurrencyAmount } {
+  const { chainId } = useActiveWeb3React()
+  const nativeCurrency = useNativeCurrency()
+  const isToken = tokenOrPair instanceof Token
```

Comment:
It's defined but not being always used. Also `isPair` is necessary

Reply:
yeah it doesn't work in that case would be cool if someone would know how to address this...